ql/src/java/org/apache/hadoop/hive/ql/exec/repl/DirCopyTask.java

```diff
@@ -70,9 +76,40 @@ private boolean createAndSetPathOwner(Path destPath, Path sourcePath) throws IOE
             destPath, sourcePath, status.getOwner(), status.getGroup(), status.getPermission());
     destPath.getFileSystem(conf).setOwner(destPath, status.getOwner(), status.getGroup());
     destPath.getFileSystem(conf).setPermission(destPath, status.getPermission());
+    setAclsToTarget(status, sourcePath, destPath);
     return createdDir;
   }
 
+  private void setAclsToTarget(FileStatus sourceStatus, Path sourcePath,
+      Path destPath) throws IOException {
+    // Check if distCp options contains preserve ACL.
+    if (isPreserveAcl()) {
+      AclStatus sourceAcls =
+          sourcePath.getFileSystem(conf).getAclStatus(sourcePath);
+      if (sourceAcls != null && sourceAcls.getEntries().size() > 0) {
+        destPath.getFileSystem(conf).removeAcl(destPath);
```

Comment:
why is this call needed?

Reply:
I followed the same semantics as DistCp, DistCp also does removeAcl first, added as part of HADOOP-16032.
The reason being if the directory  at the target  has ACCESS scope ACL and the directory at source has only Default ACL, so setAcl will not remove the ACCESS scope ACL, hence the ACL's on source & dest won't be same. So, to counter that removeACl is called first.